trillian/migrillian/core/controller.go

```diff
@@ -211,15 +211,17 @@ func (c *Controller) Run(ctx context.Context) error {
 		return err
 	}
 
+	// Note: An error in one submitter cancels the other submitters.
```

Comment:
I think that after moving the second `defer` down the code became self-explanatory, and less likely to mess the order of events. Also, the `Note` that I initially added duplicates the comment next to `cancel()` below.

Reply:
OK, I guess I'm left having to admit that I don't understand what the problem was :-)

If one submitter completes (without an error), how did the context get cancelled?

Reply:
Sorry to hear that :)
If there was no error, the context was canceled before `wg.Wait()`, which meant the other submitters were still running (likely in the process of submitting their last batches). After this fix `wg.Wait()` comes before `cancel()` which prevents the race between submitting the tailing batches and `cancel()`.

Reply:
Let me correct myself a bit. The context was canceled not because one submitter had finished, but because the `Fetcher.Run` returned. So **all** the submitters were in with `cancel()`.

Reply:
Ah, got it now, thanks.